docs/rst/tutorials/core/deployment/keys/topic_keys.rst

```diff
@@ -0,0 +1,234 @@
+
+  * :ref:`linux_binary_installation`
+  * :ref:`linux_source_installation`
+  * :ref:`docker_installation`
+
+Preparing the demo package
+--------------------------
+
+Lets start by setting up the Vulcanexus environment.
+For this, there are two possible options:
+
+#.  Running the Vulcanexus Docker image.
+
+    Run the Vulcanexus Docker image with:
+
+    .. code-block:: bash
+
+        xhost local:root
+        docker run \
+            -it \
+            ubuntu-vulcanexus:{DISTRO}-desktop
+
+    Then, within the container, source the Vulcanexus installation with:
+
+    .. code-block:: bash
+
+            source /opt/vulcanexus/{DISTRO}/setup.bash
+
+#.  Running the tutorial on the local host. For this second option,
+    it is necessary to have installed the ``vucanexus-iron-desktop`` package,
+    since this is the one that includes all the simulation tools, demos and tutorials.
+
+    Source the following file to setup the Vulcanexus environment:
+
+    .. code-block:: bash
+
+            source /opt/vulcanexus/{DISTRO}/setup.bash
+
+
+Retrieving the sources
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: bash
+
+    # Create directory structure
+    mkdir -p ~/vulcanexus_ws/src/demo_keys_cpp
+    mkdir ~/vulcanexus_ws/src/demo_keys_cpp/src
+    mkdir ~/vulcanexus_ws/src/demo_keys_cpp/msg
+    cd ~/vulcanexus_ws/src/demo_keys_cpp
+
+    # Download demo package source code
+    wget -O CMakeLists.txt https://raw.githubusercontent.com/eProsima/vulcanexus/{DISTRO}/docs/resources/tutorials/core/deployment/keys/CMakeLists.txt
+    wget -O package.xml https://raw.githubusercontent.com/eProsima/vulcanexus/{DISTRO}/docs/resources/tutorials/core/deployment/keys/package.xml
+    wget -O README.md https://raw.githubusercontent.com/eProsima/vulcanexus/{DISTRO}/docs/resources/tutorials/core/deployment/keys/README.md
+    wget -O msg/SensorDataMsg.idl https://raw.githubusercontent.com/eProsima/vulcanexus/{DISTRO}/docs/resources/tutorials/core/deployment/keys/msg/SensorDataMsg.idl
+    wget -O src/right_sensor.cpp https://raw.githubusercontent.com/eProsima/vulcanexus/{DISTRO}/docs/resources/tutorials/core/deployment/keys/src/right_sensor.cpp
+    wget -O src/left_sensor.cpp https://raw.githubusercontent.com/eProsima/vulcanexus/{DISTRO}/docs/resources/tutorials/core/deployment/keys/src/left_sensor.cpp
+    wget -O src/controller.cpp https://raw.githubusercontent.com/eProsima/vulcanexus/{DISTRO}/docs/resources/tutorials/core/deployment/keys/src/controller.cpp
+
+The resulting directory structure should be:
+
+.. code-block::
+
+    ~/vulcanexus_ws
+     ├──src
+        ├── demo_keys_cpp
+            ├── CMakeLists.txt
+            ├── README.md
+            ├── msg
+            │   └── SensorDataMsg.idl
+            ├── package.xml
+            └── src
+                ├── left_sensor.cpp
+                ├── right_sensor.cpp
+                └── controller.cpp
+
+Building the demo package
```

Comment:
I'm missing a section analyzing the most important parts of the code, such as the IDL, the CMake macro used to generate the type, the fact that there is only one sub, etc. Furthermore, I think we need some more "complexity" in the sense that it is a bit unclear what I'm gaining here over say using the same type without keys for instance. I think we also need some type of conceptual drawing of the case

Reply:
Not adressed, I thinks this is important